internal/db/db.go

```diff
@@ -189,6 +207,10 @@ func (db *DB) waitUpgrade(bootstrap bool, ext extensions.Extensions) error {
 
 	// If we are not bootstrapping, wait for an upgrade notification, or wait a minute before checking again.
 	if otherNodesBehind && !bootstrap {
+		db.statusLock.Lock()
```

Comment:
What is the reason for moving those lines to here? Wouldn't they be better placed in the `Open()` func right before the call to `db.waitUpgrade(bootstrap, ext)`? This way the status is also set for the `newSchema.Ensure(db.db)` operation inside of `waitUpgrade()`.

Reply:
And it's easier to reason about the status in case the `reverter` gets called as all the statues are set in the same func.

Reply:
My thought was that `StatusUpgrading` is used to indicate when the daemon is blocked on waiting for an upgrade, and this should signify to the user that they need to do something on their other systems. 

`waitUpgrade` always runs and spends the majority of its implementation checking the dqlite database to decide if there is an update. Only if it reaches the `select` block at the bottom does it block on expecting an update.


If we move this status up to `IsOpen`, then it will be the longest sustained status (longer than `StatusStarting`) prior to being ready, and the daemon will always be in this state on every restart, which removes any significance between `StatusUpgrading` and `StatusStarting`.

Reply:
Ok makes sense. From the status message it wasn't clear to me that this should indicate that the member is waiting on a schema update to complete (waiting on the notification). It sounds more like update process in general.

By the way, does the node on which the schema update gets executed also display the same status whilst running the update?

Reply:
That one will display that `StatusStarting` until the updates complete, since it never reaches the `upgradeCh` block.

Reply:
I think the name of that status is confusing. It's status message is "Waiting for other cluster members to update their schema versions" which is definitely not what the node that runs the updates is doing. 

I'll rename it to `StatusWaiting`.

Reply:
Alright, I've updated the status name to `StatusWaiting`. Its default message is `Database is waiting for an upgrade` and its `IsOpen` error message is `Database is waiting for an upgrade: %d cluster members have not yet received the update`.

So on bootstrap, or when starting to a fully reconciled set of schema updates & API extensions, the daemon will report `Database is still starting` until the database is available. If it detects that it cannot be available without user interaction due to a schema update not present on every node, it reports the `StatusWaiting` error instead.

Reply:
Great, this sounds feasible.